### PR TITLE
Add DomainControlValidation class with a few basic DCV methods.

### PR DESCRIPTION
**Pull request by stea-uw**
## Domain Control Validation

This implements a few methods from the [Sectigo DCV API](https://www.sectigo.com/uploads/audio/Certificate-Manager-20.1-Rest-API.html#resources-dcv-statuses).

I only added the ones I needed, hopefully this will be useful to others.

Tested locally and in our prod.

### CODE REVIEW

File: cert_manager/dcv.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+"""Define the cert_manager.dcv.DomainControlValidation class."""
+
+from requests.exceptions import HTTPError
+
+from cert_manager._endpoint import Endpoint
```
Comment (coreone) [inline cert_manager/dcv.py:6]: Please update to `from ._endpoint import Endpoint` to be consistent with the rest of the endpoints

File: cert_manager/dcv.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        https://www.sectigo.com/uploads/audio/Certificate-Manager-20.1-Rest-API.html#resources-dcv-statuses
+
+        :return list: a list of DCV statuses
+        """
+        url = self._url("validation")
+        result = self._client.get(url, params=kwargs)
+
+        return result.json()
+
+    def get_validation_status(self, domain=str):
```
Comment (coreone) [inline cert_manager/dcv.py:36]: I think this is supposed to be a type hint, so it should be `domain: str`

**Review by coreone — CHANGES_REQUESTED** [review]
Additionally, some form of unit tests for this new class should be added. Many examples can be found in the `tests` directory that should be easily adaptable to this new class.

File: cert_manager/dcv.py (hunk 2 above)
Comment (stea-uw) [reply cert_manager/dcv.py:36]: Good catch, thanks!

**Comment by stea-uw** [comment]
Thanks for the review! I've fixed the issues and added unit tests, PTAL

**Comment by stea-uw** [comment]
I ran `pre-commit --all-files` and fixed all the issues. Assuming the pre-commits are comprehensive, this should pass all the tests now.

Tested:
```
% pre-commit run --show-diff-on-failure --color=always --all-files
% python -m unittest 
```

**Review by coreone — APPROVED** [review]

**Pull request merged**